eslint.config.mjs

```diff
@@ -65,54 +125,79 @@ const config = createConfig(
   },
   {
     files: [
-      '**/jest.config.js',
-      '**/jest.environment.js',
-      '**/tests/**/*.{ts,js,tsx}',
-      '*.js',
-      '*.test.{ts,js,tsx}',
+      '**/*.{js,cjs,mjs}',
+      '**/*.test.{js,ts}',
+      '**/tests/**/*.{js,ts}',
       'scripts/*.ts',
-      'scripts/create-package/*.ts',
-      'yarn.config.cjs',
+      'scripts/create-package/**/*.ts',
     ],
     extends: [nodejs],
+    rules: {
+      // TODO: Re-enable this
+      'n/no-sync': 'off',
+      // TODO: These rules created more errors after the upgrade to ESLint 9.
+      // Re-enable these rules and address any lint violations.
+      'n/no-unsupported-features/node-builtins': 'warn',
+    },
   },
   {
-    files: ['*.test.{ts,js,tsx}', '**/tests/**/*.{ts,js,tsx}'],
```

Comment:
We don't have a `'**/tests/` directory in the design system monorepo so this can be adjusted to reflect that